landoui/treestatus.py

```diff
@@ -211,14 +211,24 @@ def treestatus_tree(tree: str):
     """Display the log of statuses for an individual tree."""
     api = TreestatusAPI.from_environment()
 
-    logs_response = api.request("GET", f"trees/{tree}/logs")
-    logs = logs_response.get("result")
+    try:
+        logs_response = api.request("GET", f"trees/{tree}/logs")
+
+        # We got a response, if this is empty is it an error.
+        logs = logs_response.get("result")
+        detail = "empty response from Lando"
+    except LandoAPIError as exc:
+        if not exc.detail:
+            raise exc
```

Comment:
The status code is returned directly from Lando API. This should be used to determine the status code that is returned by Lando UI. There is detail being lost in this implementation. At the very least, we can capture the `404` that we already know is being triggered and handle that specific flow, and continue to log other exceptions to see if they need to be handled in the future in a different way.

Reply:
Re-raising the `LandoAPIError` when we don't get a response in the `ProblemException` format is how this error is handled in other parts of Lando. I did update this to be more generic about API errors received from the logs endpoint, including using the API response code in the UI redirect.